src/odemis/gui/cont/tabs.py

```diff
@@ -4352,8 +4353,14 @@ def _create_needed_tabs(self, tab_defs, main_frame, main_data):
                     max_prio = priority
                     default_tab = tab
                 tabs.append(tab)
-            else:
-                tab_def["button"].Hide()  # Hide the tab button
+                assert tab.button not in buttons
+                buttons.add(tab.button)
+
+        # Hides the buttons which are not used
+        for tab_def in tab_defs:
+            b = tab_def["button"]
+            if b not in buttons:
+                b.Hide()
```

Comment:
Why did you move this outside the other loop?

Reply:
Because now buttons can be shared (see commit message). So we cannot immediately hide a button if the tab is not shown. We have to check that no tab uses this button.